### PR TITLE
Reenable ocamlformat in backend subdirectories

### PR DESCRIPTION
**Pull request by gretay-js**
`ocamlformat` was accidently disabled in subdirectories of `backend` by the change in #631.
This PR fixes it and cleans up unnecessarily duplicates of .ocamlformat files in subdirectories of backend, listing the enabled directories in `backend/.enable-ocamlformat` instead. 

The second commit is automatically generated by `make fmt` to fix up recent commits.

### CODE REVIEW

File: backend/.ocamlformat-enable [hunk 1]
```diff
@@ -1,2 +1,8 @@
 cmm_helpers.ml
 cmm_helpers.mli
+cfg/*.ml
+cfg/*.mli
+asm_targets/*.ml
+asm_targets/*.mli
+debug/dwarf/*.ml
+debug/dwarf/*.mli
```
Comment (mshinwell) [inline backend/.ocamlformat-enable:8]: Does this need to list the subdirectories of `dwarf/` too?

**Review by mshinwell — CHANGES_REQUESTED** [review]

File: backend/.ocamlformat-enable (hunk 1 above)
Comment (mshinwell) [reply backend/.ocamlformat-enable]: actually, could we change this to cover `debug/*.ml{,i}` even if there are no files matching that at present?

Comment (gretay-js) [reply backend/.ocamlformat-enable]: Do you mean to cover `debug/dwarf/*.ml{,i}` ?

Comment (mshinwell) [reply backend/.ocamlformat-enable]: No, I mean directly under `debug/`, since we will add files in there in due course.

**Comment by mshinwell** [comment]
@gretay-js could you rebase this?

File: backend/.ocamlformat-enable [hunk 2]
```diff
@@ -1,2 +1,12 @@
 cmm_helpers.ml
 cmm_helpers.mli
+cfg/*.ml
```
Comment (xclerc) [inline backend/.ocamlformat-enable:3]: I wonder whether we could simplify
things by using patterns like `**/*.ml`
and `**/*.mli`.

**Review by xclerc — APPROVED** [review]

File: backend/.ocamlformat-enable (hunk 2 above)
Comment (gretay-js) [reply backend/.ocamlformat-enable]: It works, thank you! and now there is a new commit for you to review.

**Review by mshinwell — APPROVED** [review]

**Pull request merged**